src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CertificatePolicy.cs

```diff
@@ -287,41 +290,45 @@ private static bool CheckExplicitAnyPolicy(ISet<string> declaredPolicies)
             return declaredPolicies.Remove(Oids.AnyCertPolicy);
         }
 
-        private static int ReadInhibitAnyPolicyExtension(X509Extension extension)
+        private static int ReadInhibitAnyPolicyExtension(byte[] rawData)
         {
-            AsnReader reader = new AsnReader(extension.RawData, AsnEncodingRules.DER);
+            AsnReader reader = new AsnReader(rawData, AsnEncodingRules.DER);
             int inhibitAnyPolicy;
             reader.TryReadInt32(out inhibitAnyPolicy);
             reader.ThrowIfNotEmpty();
             return inhibitAnyPolicy;
         }
 
-        private static void ReadCertPolicyConstraintsExtension(X509Extension extension, CertificatePolicy policy)
+        private static void ReadCertPolicyConstraintsExtension(byte[] rawData, CertificatePolicy policy)
         {
             PolicyConstraintsAsn constraints = PolicyConstraintsAsn.Decode(
-                extension.RawData,
+                rawData,
                 AsnEncodingRules.DER);
 
             policy.RequireExplicitPolicyDepth = constraints.RequireExplicitPolicyDepth;
             policy.InhibitMappingDepth = constraints.InhibitMappingDepth;
         }
 
-        private static ISet<string> ReadExtendedKeyUsageExtension(X509Extension extension)
+        private static ISet<string> ReadExtendedKeyUsageExtension(byte[] rawData)
         {
-            X509EnhancedKeyUsageExtension ekusExtension = (X509EnhancedKeyUsageExtension)extension;
             HashSet<string> oids = new HashSet<string>();
 
-            foreach (Oid oid in ekusExtension.EnhancedKeyUsages)
+            AsnReader reader = new AsnReader(rawData, AsnEncodingRules.BER);
```

Comment:
This one should be DER

Reply:
updated. I probably look at wrong place for example. Do you have  example where BER would fail?  If easy enough I can add some new test.

Reply:
BER is more permissive, it'll never fail where DER succeeds.  So this is just "read in strict mode, because certs say only strict mode is valid".